src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java

```diff
@@ -0,0 +1,520 @@
+        {
+            final int numOfAnimals = searchForAnimals().size();
+            final int maxAnimals = getOwnBuilding().getBuildingLevel() * getMaxAnimalMultiplier();
+
+            return numOfAnimals > maxAnimals;
+        }
+        return true;
+    }
+
+    /**
+     * Sets the tool as held item.
+     *
+     * @param toolType the {@link ToolType} we want to equip
+     * @return true if the tool was equipped.
+     */
+    public boolean equipTool(final ToolType toolType)
+    {
+        if (getToolSlot(toolType) != -1)
+        {
+            worker.setHeldItem(getToolSlot(toolType));
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Gets the slot in which the Tool is in.
+     *
+     * @return slot number.
+     */
+    private int getToolSlot(final ToolType toolType)
+    {
+        if (getOwnBuilding() != null)
+        {
+            final int slot = InventoryUtils.getFirstSlotOfItemHandlerContainingTool(new InvWrapper(getInventory()), toolType,
+              TOOL_LEVEL_WOOD_OR_GOLD, getOwnBuilding().getMaxToolLevel());
+
+            if (slot == -1)
+            {
+                checkForToolOrWeapon(toolType);
+            }
+            return slot;
+        }
+        return -1;
+    }
+
+    /**
+     * Sets the {@link ItemStack} as held item or returns false.
+     *
+     * @param itemStack the {@link ItemStack} to equip.
+     * @return true if the item was equipped.
+     */
+    public boolean equipItem(final ItemStack itemStack)
+    {
+        if (checkIfRequestForItemExistOrCreateAsynch(itemStack))
+        {
+            worker.setHeldItem(getItemSlot(itemStack.getItem()));
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Gets the slot in which the inserted item is in. (if any).
+     *
+     * @param item The {@link Item} to check for.
+     * @return slot number -1 if not in INV.
+     */
+    public int getItemSlot(final Item item)
+    {
+        return InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(getInventory()), item, 0);
+    }
+
+    /**
+     * Butcher an animal.
+     *
+     * @param animal the {@link EntityAnimal} we are butchering
+     */
+    private void butcherAnimal(final EntityAnimal animal)
+    {
+
+        worker.setLatestStatus(new TextComponentTranslation(TranslationConstants.COM_MINECOLONIES_COREMOD_STATUS_HERDER_BUTCHERING));
+
+        if (!walkingToAnimal(animal) && worker.getHeldItemMainhand() != null && animal != null)
+        {
+            new DamageSource(worker.getName());
+
+            worker.swingArm(EnumHand.MAIN_HAND);
+            animal.attackEntityFrom(new DamageSource(worker.getName()), (float) BUTCHERING_ATTACK_DAMAGE);
+
+            worker.getHeldItemMainhand().damageItem(1, animal);
+        }
+    }
+
+    /**
+     * Get breeding item for animal.
+     *
+     * @return the {@link Item} needed for breeding the animal.
+     */
+    public Item getBreedingItem()
```

Comment:
Make this: public Abstract ItemStack getBreedingStack(), and have it implemented on the actualy AI class.

Reply:
yep true, didn't think of that sorry